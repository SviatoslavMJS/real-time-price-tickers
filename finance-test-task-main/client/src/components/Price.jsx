import s from '../styles/Price.module.css';

function Price(props) {
    return (
        <div className={s.price}>
            {'$ ' + props.price}
        </div>
    )
}

export default Price;