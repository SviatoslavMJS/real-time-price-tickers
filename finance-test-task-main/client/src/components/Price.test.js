import Price from "./Price.jsx";
import renderer from 'react-test-renderer';

test("On change prop price in Price text must change from 279.29 to 89", () => {
    const component = renderer.create(
      <Price price={"279.29"} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.component = "89";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.component = "279.29";

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
});