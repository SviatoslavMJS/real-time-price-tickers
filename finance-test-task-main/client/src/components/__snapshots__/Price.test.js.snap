// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On change prop price in Price text must change from 279.29 to 89 1`] = `
<div
  className="price"
>
  $ 279.29
</div>
`;

exports[`On change prop price in Price text must change from 279.29 to 89 2`] = `
<div
  className="price"
>
  $ 279.29
</div>
`;

exports[`On change prop price in Price text must change from 279.29 to 89 3`] = `
<div
  className="price"
>
  $ 279.29
</div>
`;
